spec/rubocop/cop/style/mutable_constant_spec.rb

```diff
@@ -157,6 +196,32 @@
           RUBY
         end
       end
+
+      context 'when using shareable_constant_value' do
+        it_behaves_like 'freezing all constants', '# shareable_constant_value: literal'
+        it_behaves_like 'freezing all constants', '# shareable_constant_value: experimental_everything'
+        it_behaves_like 'freezing all constants', '# shareable_constant_value: experimental_copy'
+        it_behaves_like 'not freezing all constants', '# shareable_constant_value: none'
+      end
+
+      it 'raises offense when shareable_constant_value is specified as an inline comment' do
+        expect_offense(<<~RUBY)
+          X = [1, 2, 3] # shareable_constant_value: literal
+              ^^^^^^^^^ Freeze mutable objects assigned to constants.
+        RUBY
```

Comment:
Did you verify what Ruby does for inline comments? How does it affect subsequent constants? I'm not sure what it does, and [the doc](https://docs.ruby-lang.org/en/3.0.0/doc/syntax/comments_rdoc.html#label-Scope) doesn't say. Do you feel like opening a bug report or proposing a PR for Ruby about this, or should I do it?
Also, the comments are actually scoped; current code does not take that into account. This limitation should be documented.

Reply:
yes @marcandre - verified in my local. The above test is  the observed behaviour.
``` ruby
A = %w[anything here is fine even non-literals] # shareable_constant_value: literal 
puts A.frozen? # prints false
```

I can open a pr in ruby for the docs. And yes, that was a good catch, i first went to the docs after your review comment but not help there.

Reply:
Could you test how it affects constants that follow please (both in Ruby and in this PR)?

Reply:
Verified it in ruby 3.0 and updated the test in PR. 
**inline comment is not recognised.**

Reply:
In addition, I was trying to raise PR for this in ruby documentation - any tips on which repo and where I should go about this? https://github.com/ruby/ruby/blob/master/doc/contributing.rdoc is not very clear where to contribute for docs. https://github.com/ruby/docs.ruby-lang.org - this led me no where as well.

Reply:
I guess it's ok; magic comments need always be alone on a line